### ISSUE TITLE
Create simple filesystem

### ISSUE THREAD

**Issue by iankronquist** [iankronquist/kernel-of-truth#34]
* osdev wiki [recommends FAT](http://wiki.osdev.org/File_Systems#.22Beginners.22_filesystems) because it's simple.
* ToaruOS implements EXT2
* Fat is copyrighted. Ich. I would prefer to not have MS copyrighted code in my kernel.

Documentation:
* [FAT spec from MS](https://staff.washington.edu/dittrich/misc/fatgen103.pdf)
* [Looks kind of like an ext2 spec](http://www.nongnu.org/ext2-doc/ext2.html)
* [Paper on ext2](http://web.mit.edu/tytso/www/linux/ext2intro.html)
* [Summary of ext2](http://uranus.chrysocome.net/explore2fs/es2fs.htm)

**Comment by Herbstein** [comment]
We should probably also consider the SFS file system since it is the easiest to implement, according to [wiki.osdev.org](http://wiki.osdev.org/File_Systems#File_systems_for_OSDevers). This isn't the easiest task either way, but implementing the easiest file system seems like an attractive option.

**Comment by iankronquist** [comment]
@Herbstein I'm game for that. We should probably do the hard disk drivers first though.

**Comment by Herbstein** [comment]
Definitely. I must admit that I'm in deep water contributing to this project, so you might be pulling a little more than I'm pushing. I have started reading the Tanenbaum os book, Modern Operating Systems. I might ask some dumb questions, hope you'll bear with me.

**Comment by iankronquist** [comment]
Oh, don't worry, I'm the king of dumb questions around here. I'm sure you know about the osdev forums and the #osdev IRC channel on freenode. I'm known as radens over there.

Tanenbaum is a good place to start. I think I put a bunch of other links in the README.
